design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,400 @@
+    telephone: String! @doc(description: "Company's phone number.")
+}
+
+type CompanyAdmin @doc(description: "Company Administrator (Customer with corresponding privileges) output data schema.") {
+    id: ID! @doc(description: "Company Administrator's id.")
+    email: String! @doc(description: "Company Administrator email address.")
+    firstname: String! @doc(description: "Company Administrator first name.")
+    lastname: String! @doc(description: "Company Administrator last name.")
+    job_title: String @doc(description: "Company Administrator job title.")
+    gender: Int @doc(description: "Company Administrator gender.")
+}
+
+type CompanyAddressRegion @doc(description: "Output data schema for a region data in Company address.") {
+    region_id: ID! @doc(description: "Unique region identifier. Required for certain countries.")
+    region: String @doc(description: "State or province name.")
+    region_code: String @doc(description: "Region code.")
+}
+
+type CompanySalesRepresentative @doc(description: "Company sales representative information output data schema.") {
+    email: String! @doc(description: "Sales representative email address.")
+    firstname: String! @doc(description: "Sales representative first name.")
+    lastname: String! @doc(description: "Sales representative last name.")
+}
+
+type CompanyUsers @doc(description: "Output data schema for an object returned by a Company users search query.") {
+    items: [Customer] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "Pagination meta data.")
+}
+
+type CompanyRoles @doc(description: "Output data schema for an object returned by a Company roles search query.") {
+    items: [CompanyRole] @doc(description: "A list of company roles that match the specified search criteria.")
+    total_count: Int @doc(description: "The total number of objects matching the specified filter.")
+    page_info: SearchResultPageInfo @doc(description: "Pagination meta data.")
+}
+
+type CompanyRole @doc(description: "Company role output data schema returned in response to a query by Role id.") {
+    id: ID! @doc(description: "Role id.")
+    name: String! @doc(description: "Role name.")
+    users_count: Int @doc(description: "Total number of Users with such Role within Company Hierarchy.")
+    permissions: [String] @doc(description: "A list of permission resources defined for a Role.")
+}
+
+type CompanyAclResource @doc(description: "Output data schema for an object with Role permission resource information.") {
+    id: ID! @doc(description: "ACL resource id.")
+    text: String! @doc(description: "ACL resource label.")
+    sortOrder: Int! @doc(description: "ACL resource sort order.")
+    children: [CompanyAclResource!] @doc(description: "An array of sub-resources.")
```

Comment:
Similar to https://github.com/magento/architecture/pull/366/files#r403141125 - we should strive to avoid unbounded recursive types like this

Reply:
The list of permissions contains very static, not deep tree. Decided to go with tree structure.